src/prf_expand.rs

```diff
@@ -108,3 +108,20 @@ impl PrfExpand<([u8; 96], [u8; 32], [u8; 4])> {
     pub const SAPLING_ZIP32_CHILD_NON_HARDENED: Self = Self::new(0x12);
 }
 with_inputs!(a, A, b, B, c, C);
+
+impl PrfExpand<([u8; 32], [u8; 4])> {
+    pub const REGISTERED_ZIP32_CHILD: Self = Self::new(0xAC);
+
+    /// Expands the given secret key in this domain, with additional `lead`
+    /// and `tag` inputs.
+    pub fn with_tag(
+        self,
+        c_par: &[u8],
+        sk_par: &[u8; 32],
+        i: &[u8; 4],
+        lead: &[u8],
```

Comment:
Why is the lead variable length? I thought that this was going to just be either a single byte, or maybe 4 bytes?

Reply:
It can be 0 or 1 bytes. It will be 0 bytes if we are in the case $\mathsf{tag} = []$ and not $\mathsf{full\_width\_leaf}$, so that in that case child key derivation is the same for Registered keys as for the existing use by Orchard (and by the deprecated Arbitrary keys). That's not essential; I considered the option of always encoding this byte. But it has the advantage that if an implementation doesn't require tags or full-width leaves, then its $\mathsf{CKDh}$ doesn't need to change. I think it is also easier to think about the new $\mathsf{CKDh}$ as a strict generalization.

Reply:
If it can be zero or 1 byte, it should be `Option<u8>`, not a slice.

Reply:
Done.